front/app/components/EsriMap/index.tsx

```diff
@@ -9,101 +9,137 @@ import Layer from '@arcgis/core/layers/Layer';
 import Graphic from '@arcgis/core/Graphic';
 import { Box } from '@citizenlab/cl2-component-library';
 import Fullscreen from '@arcgis/core/widgets/Fullscreen';
+import Point from '@arcgis/core/geometry/Point';
 
 // hooks
 import useAppConfiguration from 'api/app_configuration/useAppConfiguration';
 
 // utils
 import { getDefaultBasemap } from './utils';
 import { isNil } from 'utils/helperUtils';
+import { EsriUiElement } from './types';
 
 type Props = {
-  center?: GeoJSON.Point | null;
   height?: string;
   width?: string;
-  zoom?: number;
-  maxZoom?: number;
+  initialData?: InitialData;
   layers?: Layer[];
-  onClick?: (event: any, mapView: MapView) => void;
   graphics?: Graphic[];
-  showFullscreenOption?: boolean;
 };
 
-const EsriMap = ({
-  center,
-  zoom,
-  maxZoom,
-  layers,
-  onClick,
-  height,
-  width,
-  graphics,
-  showFullscreenOption,
-}: Props) => {
+type InitialData = {
+  initialCenter?: GeoJSON.Point | null;
+  initialZoom?: number;
+  initialMaxZoom?: number;
+  initialUiElements?: EsriUiElement[];
+  initialShowFullscreenOption?: boolean;
+  initialOnClick?: (event: any, mapView: MapView) => void;
+  initialOnHover?: (event: any, mapView: MapView) => void;
+};
+
+const EsriMap = ({ height, width, layers, graphics, initialData }: Props) => {
   const { data: appConfig } = useAppConfiguration();
+  const globalMapSettings = appConfig?.data.attributes.settings.maps;
+
+  const [map, setMap] = useState<Map | null>(null);
+  const [mapView, setMapView] = useState<MapView | null>(null);
+
+  const initialValuesLoaded = useRef(false);
+
+  // On initial render, create a new map and mapView and save them to state variables
+  useEffect(() => {
+    const newMap = new Map();
+    setMap(newMap);
+
+    setMapView(
+      new MapView({
+        container: 'esriMap', // Reference to DOM node that will contain the view
+        map: newMap,
+      })
+    );
+  }, []);
 
+  // Load initial map configuration data that was passed in.
+  // Note: This data is static and will not change.
   useEffect(() => {
-    // Get global map settings from the app config
-    const globalMapSettings = appConfig?.data.attributes.settings.maps;
-
-    // Create a new map and map view
-    const esriMap = new Map();
-
-    const mapView = new MapView({
-      container: 'esriMap', // Reference to DOM node that will contain the view
-      map: esriMap,
-      zoom: zoom || globalMapSettings?.zoom_level,
-      center: !isNil(center)
-        ? [center.coordinates[0], center.coordinates[1]]
-        : undefined,
-      constraints: {
-        maxZoom: maxZoom || 22,
+    if (!initialValuesLoaded.current && initialData && mapView && map) {
```

Comment:
I used a ref here just to make sure we only load these values once - but I'm not sure if that's good practice? It seems to do what I'd like it to do at least :thinking: